app/controllers/template_invocations_controller.rb

```diff
@@ -7,15 +7,14 @@ def show
     @template_invocation_task = ForemanTasks::Task.find(params[:id])
     @template_invocation = @template_invocation_task.locks.where(:resource_type => 'TemplateInvocation').first.try(:resource)
     @host = @template_invocation_task.locks.where(:resource_type => 'Host::Managed').first.try(:resource)
-
-    respond_to do |format|
-      format.html
-      format.js do
-        @lines = @template_invocation_task.main_action.output
-        if params[:since]
-          @lines = @lines.select { |o| o['timestamp'].to_f > params[:since].to_f }
-        end
-      end
+    @auto_refresh = @template_invocation_task.pending?
+    @lines = @template_invocation_task.main_action.live_output
+    @since = params[:since].to_f if params[:since].present?
+    if @since
+      @line_counter = params[:line_counter].to_i
```

Comment:
it is: in case the timestamp at some line was nil, you don't filter anything, therefore you always start from 0. I added this at the beginning of the PR, where not all lines had timestamps. I can revisit that: and make sure the timestamps are always present, what about that?